vars/githubWorkflowRun.groovy

```diff
@@ -0,0 +1,109 @@
+
+  githubWorkflowRun( args map )
+
+  Args:
+    repo - repositiry owner and name 
+    workflow - workflow file name
+    ref - reference ( branch, tag or hash)
+    parameters - map with parameters to pass to the workflow 
+
+  Returs:
+    runInfo - information about
+
+  Example:
+     script {
+       Map args = [
+           repo: "owner/repository",
+           workflow: "build.yml", 
+           ref: "main",
+           parameters: [
+             path: "filebeat"
+           ],
+           credentialsId: "github-workflow-token",
+           ghVersion: "2.1.0"]
+       int runId = githubWorkflowRun.triggerGithubActionsWorkflow(args)
+       Map runInfo = githubWorkflowRun.getWorkflowRun(args + [runId: runId]) 
+   }
+*/
+
+import groovy.json.JsonSlurperClassic 
+
+def call(Map args = [:]) {
+    try {
+        int buildTimeLimit = args.get('buildTimeLimit', 1)
+        int runId = triggerGithubActionsWorkflow(args)
+        Date startDate = new Date()
+        while(startDate.time  + buildTimeLimit*3600000 -  new Date().time > 0) {
+            Map runInfo = getWorkflowRun(args + [runId: runId])
+            if (runInfo.status == "completed") return runInfo
+            Thread.sleep(300000); // sleep 5 min
+        }
+        throw new Error("Build time out")
+    } catch(error) {
+        log(level: 'ERROR', text: "githubWorkflowRun: Failed to run workflow:\n${error.message} ")
+    }
+    return null
+}
+
+int triggerGithubActionsWorkflow(Map args = [:]) {
+    if (!args.repo || !args.workflow) {
+        throw new IllegalArgumentException("repo: '${args.repo}', workflow: '${args.workflow}'")
+    }
+    def ghVersion = args.get("ghVersion", "2.1.0")
+    def ref = args.get("ref", "master")
+    def runner = args.get("runner", "ubuntu-latest")
+    def runId = "${ref}-${new Date().getTime()}-${env.BUILD_ID}"
+    def parameters = args.get("parameters", [:])
+    def inputs = (parameters + [id: runId, runner: runner]).collect{ "${it}" }
+    gh(command: "workflow run ${args.workflow}", credentialsId: args.credentialsId,
+        version: ghVersion, forceInstallation: true,
+        flags: [repo: args.repo, ref: ref, field: inputs ])
+    Thread.sleep(30000)
+    def runsText = gh(command: "run list", credentialsId: args.credentialsId,
+        version: ghVersion, forceInstallation: false,
+        flags: [repo: args.repo, workflow: args.workflow, limit: 100])
+    def runIds = []
+    for(def s: runsText.split("\n")) {
+        def r = s.split(/\s+/)
+        if (r[-1] == "0m"|| r[-1] == "1m") runIds.add(r[-3]) 
+    }
+    for(def run: runIds) {
+        def runText=gh(command: "run view -v ${run}", credentialsId: args.credentialsId,
+            version: ghVersion, forceInstallation: false, flags: [repo: args.repo])
+        if (runText.split("\n").find { it ==~ /^\s+\S\s+Run ID ${runId}\s*$/} ) {
+            return run as int
+        }
+    }
+    return 0
+}
+
+Map getWorkflowRun(Map args = [:]) {
```

Comment:
If this is a private method then the arguments validation could be delegated to the function that consumes this method